framework/src/base/BlockRestrictable.C

```diff
@@ -27,6 +27,8 @@ InputParameters validParams<BlockRestrictable>()
   // Add the user-facing 'block' input parameter
   params.addParam<std::vector<SubdomainName> >("block", "The list of block ids (SubdomainID) that this object will be applied");
 
+  params.addPrivateParam<bool>("delay_initialization", false);
```

Comment:
Parameters _are_ documented.  They show up in `--dump`, Peacock and the syntax web-pages... EXCEPT for the private ones.